recipes/osx-sysroot/meta.yaml

```
@@ -0,0 +1,71 @@
+{% set name = "osx-sysroot" %}
+{% set version = "1" %}
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+build:
+  number: 0
+
+outputs:
+  - name: macosx-sdk
+    version: {{ _MACOSX_SDK_VERSION_ }}
+    script: build_sdk.sh
+    build:
+      noarch: generic
+      # weigh down newer sdks
+      track_features:                                 # [is_default_sdk == "no"]
+        - macosx-sdk_{{ _MACOSX_SDK_VERSION_ }}       # [is_default_sdk == "no"]
+    requirements:
+      run_constrained:
+        # we cannot build for the future!
+        - sysroot_osx-64 <={{ _MACOSX_SDK_VERSION_ }}
+        - sysroot_osx-arm64 <={{ _MACOSX_SDK_VERSION_ }}
+    test:
+      commands:
+        - "[[ ${MACOSX_SDK_VERSION} == '{{ _MACOSX_SDK_VERSION_ }}' ]]"
+        - test -d "${MACOSX_SDK_CACHE_DIR}/MacOSX${MACOSX_SDK_VERSION}.sdk"
+        - test -d "${SDKROOT}"
+    about:
+      home: https://github.com/conda-forge/osx-sysroot-feedstock
+      summary: package to download MacOSX SDKs
+      license: BSD-3-Clause
+      license_family: BSD
+      license_file: LICENSE
+
+  - name: sysroot_{{ cross_target_platform }}
```

```suggestion
  - name: macosx_deployment_target_{{ cross_target_platform }}
```
?

sysroot does not make sense for me.

---

Even this is no fun

```
build:
  - {{ sysroot('default') }}
  - macosx_deployment_target  # [osx]
```

---

`compiler('c')` is expanded to `{{ c_compiler }}_{{ target_platform }} {{ c_compiler_version }}.*` where `c_compiler` expands to `gcc` on linux and `clang` on macos. We can do the same.

---

Agreed, that's what I meant with `sysroot_name` in https://github.com/conda/conda-build/issues/4981 (or whatever that jinja function gets called; not sure how well `macosx_deployment_target` would fit with `c_stdlib`, but that's a bikeshedding detail). The important thing is that we should be able to name this differently per platform.

---

Yep for the sysroot jinja2 the magic expansion works ofc. However, if we separate the deployment target from the SDK, then we need two things in build for osx, but only one for linux. This in effect means we have to use a selector as I noted above.

---

Would prefer simpler user syntax over things

---

name changed here!